src/Hadolint/Rules.hs

```diff
@@ -484,7 +484,7 @@ pipVersionPinned = instructionRule code severity message check
 findPackages :: [String] -> [String]
 findPackages = takeWhile (not . isEnd) . dropWhile isFlag
   where
-    isEnd word = isFlag word || word == "&&"
+    isEnd word = isFlag word || word `elem` ["&&", "||", ";"]
```

Comment:
What about pipe `|` ?

Reply:
out ops, forgot about it.

Aside: do you actually use a single pipe in bash inside docker? I've never written one, let alone in a dockerfile?

Reply:
I have it usually in this scenario

```Dockerfile
RUN curl -sSL https://get.docker.com/builds/Linux/x86_64/docker-${DOCKER_VERSION}.tgz \
  | tar -xzv --strip-components=1 -C /bin \
  && chmod +x /bin/docker-* \
  && ln -s /bin/docker /bin/docker-${DOCKER_VERSION}
```

Reply:
ok, but not in a `pip install` right? I don't think it will hurt leaving it there, so no worries

Reply:
Yes, but I am not a heavy user of python projects. I believe there is somebody who will find it useful :smile: